src/main/java/seedu/address/logic/commands/StackUndoRedo.java

```diff
@@ -29,15 +25,36 @@ public StackUndoRedo() {
      * if {@code command} is not of type {@code UndoCommand} or {@code RedoCommand}.
      */
     public void push(Command command) {
-        if (!(command instanceof UndoCommand) && !(command instanceof RedoCommand)) {
+
+        if (isNotRedoableCommand(command)) {
+            return;
+        }
+
+        if (isNotUndoRedoCommand(command)) {
             redoStack.clear();
         }
 
+        undoStack.add((RedoableCommand) command);
+    }
+
+    /**
+     * Returns true if the command is not redoable or undoable.
+     */
+    public boolean isNotRedoableCommand(Command command) {
         if (!(command instanceof RedoableCommand)) {
-            return;
+            return true;
         }
+        return false;
+    }
 
-        undoStack.add((RedoableCommand) command);
+    /**
+     * Returns true if the command is either an undo command or a redo command.
+     */
+    public boolean isNotUndoRedoCommand(Command command) {
+        if (!(command instanceof UndoCommand) && !(command instanceof RedoCommand)) {
+            return true;
+        }
+        return false;
```

Comment:
Why refactor this into a different method?

Reply:
^, and even though it's good to use SLAP here, you would not need two methods. Just one is enough.